Fix issues for fairseq model handling and VC when speaker_wav is passed

hanasim opened the pull request:
When we use the model "tts_models/eng/fairseq/vits", got the following error.

Error in text to speech: cannot access local variable 'dataset' where it is not associated with a value

at line ..../.conda/envs/venv/lib/python3.11/site-packages/TTS/utils/manage.py", line 304, in _set_model_item
    model_full_name = f"{model_type}--{lang}--{dataset}--{model}"

hanasim committed:
Extract dataset and model for fairseq models

When we use the model "tts_models/eng/fairseq/vits", got the following error.

Error in text to speech: cannot access local variable 'dataset' where it is not associated with a value

at line ..../.conda/envs/venv/lib/python3.11/site-packages/TTS/utils/manage.py", line 304, in _set_model_item
    model_full_name = f"{model_type}--{lang}--{dataset}--{model}"

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/coqui-ai/TTS?pullRequest=3765) <br/>All committers have signed the CLA.

hanasim committed:
Pass speaker_wav argument to tts_to_file in tts_with_vc method

When `tts_with_vc_to_file` is invoked with only `speaker_wav` and not `speaker`, it was throwing the error.

Error in text to speech: Model is multi-speaker but no `speaker` is provided.

```
 File ".../.conda/envs/venv/lib/python3.11/site-packages/TTS/api.py", line 416, in tts_with_vc
    self.tts_to_file(
  File ".../.conda/envs/venv/lib/python3.11/site-packages/TTS/api.py", line 333, in tts_to_file
    self._check_arguments(speaker=speaker, language=language, speaker_wav=speaker_wav, **kwargs)
  File ".../.conda/envs/venv/lib/python3.11/site-packages/TTS/api.py", line 228, in _check_arguments
    raise ValueError("Model is multi-speaker but no `speaker` is provided.")
ValueError: Model is multi-speaker but no `speaker` is provided.
ERROR:app.std.app_logger:Error in text to speech: Model is multi-speaker but no `speaker` is provided.

eginhard commented:
This repo is not maintained anymore. The issue is already fixed in [our fork](https://github.com/idiap/coqui-ai-TTS), available via `pip install coqui-tts`
